arch/arm/CMakeLists.txt

```diff
@@ -1,11 +1,12 @@
 zephyr_cc_option_ifdef(CONFIG_LTO -flto)
 
-set(ARCH_FOR_cortex-m0     armv6-m     )
-set(ARCH_FOR_cortex-m0plus armv6-m     )
-set(ARCH_FOR_cortex-m3     armv7-m     )
-set(ARCH_FOR_cortex-m4     armv7e-m    )
-set(ARCH_FOR_cortex-m23    armv8-m.base)
-set(ARCH_FOR_cortex-m33    armv8-m.main)
+set(ARCH_FOR_cortex-m0        armv6-m         )
+set(ARCH_FOR_cortex-m0plus    armv6-m         )
+set(ARCH_FOR_cortex-m3        armv7-m         )
+set(ARCH_FOR_cortex-m4        armv7e-m        )
+set(ARCH_FOR_cortex-m23       armv8-m.base    )
+set(ARCH_FOR_cortex-m33       armv8-m.main+dsp)
```

Comment:
Why not do it the other way around? 
```
ARCH_FOR_cortex-m33       armv8-m.main
ARCH_FOR_cortex-m33+dsp armv8-m.main+dsp
```

Reply:
@carlescufi mcpu=cortex_m33 is _incompatible_ with march=armv8-m.main; GCC compiler gives a warning:

>cc1.exe: warning: switch -mcpu=cortex-m33 conflicts with -march=armv8-m.main switch

Reply:
ok understood

Reply:
Is this related to the bug we the current ARM toolchain?

Reply:
@galak I am not sure about it.
AFAIK, the "bug" is the conflict with the SVC instruction support (related PR; #7445)

GCC tool-chain does not support the -march=armv8-m.main+nodsp option yet. However,
as of its latest release (2008 7 update 2, see Release notes) the tool-chain has started supporting the -mcpu=cortex-m33+nodsp option.

This makes me think that the tool-chain assumes, now, the following:
- cortex-m33+nodsp compatible with: armv8-m.main
- cortex-m33 compatible with armv8-m.main+dsp

 This options allow to build with or without the DSP support, by solely differentiating on the -mcpu option.

Reply:
I think there was the issue that the 2017 gcc release didn't support cortex-m33+nodsp, and that cortex-m33 implied the dsp options.